src/main/fc/rc.c

```diff
@@ -537,6 +549,18 @@ static FAST_CODE void processRcSmoothingFilter(void)
             *dst = rxDataToSmooth[i];
         }
     }
+
+    // for ANGLE and HORIZON, smooth rcDeflection on pitch and roll
+#ifdef USE_ACC
+    if (flightModeFlags == ANGLE_MODE || flightModeFlags == HORIZON_MODE) {
+        for (int axis = FD_ROLL; axis < FD_YAW; axis++) {
+            if (rcSmoothingData.filterInitialized) {
+                rcDeflectionSmoothed[axis] = pt3FilterApply(&rcSmoothingData.filterDeflection[axis], rcDeflection[axis]);
+            }
+        }
+    }
+#endif
+
```

Comment:
You can pull the conditional statement out of the loop to prevent multiple evaluations:

```c
   // for ANGLE and HORIZON, smooth rcDeflection on pitch and roll
#ifdef USE_ACC
    if ((flightModeFlags == ANGLE_MODE || flightModeFlags == HORIZON_MODE) && rcSmoothingData.filterInitialized) {
        for (int axis = FD_ROLL; axis < FD_YAW; axis++) {
            rcDeflectionSmoothed[axis] = pt3FilterApply(&rcSmoothingData.filterDeflection[axis], rcDeflection[axis]);
        }
    }
#endif
```

And as far as I could check this in the codebase real quick, the enabled flight modes need to be checked with the `FLIGHT_MODE()` macro because the flight mode flags are bit masks:
```c
   // for ANGLE and HORIZON, smooth rcDeflection on pitch and roll
#ifdef USE_ACC
    if ((FLIGHT_MODE(ANGLE_MODE) || FLIGHT_MODE(HORIZON_MODE)) && rcSmoothingData.filterInitialized) {
        for (int axis = FD_ROLL; axis < FD_YAW; axis++) {
            rcDeflectionSmoothed[axis] = pt3FilterApply(&rcSmoothingData.filterDeflection[axis], rcDeflection[axis]);
        }
    }
#endif
```
Otherwise we could potentially run into problems where angle or horizon modes are enabled but don't get detected.

Reply:
Thanks for the correction and advice.
Resolved here
https://github.com/betaflight/betaflight/pull/10774/commits/a076f8d3bd06c465871c0432753ce633078387c4